x/clp/keeper/calculations.go

```diff
@@ -286,6 +287,12 @@ func CalculatePoolUnits(oldPoolUnits, nativeAssetBalance, externalAssetBalance,
 		return sdk.ZeroUint(), sdk.ZeroUint(), types.ErrAsymmetricAdd
 	}
 
+	ratioDiff := R.Quo(A).Sub(r.Quo(a)).Abs()
+
+	if ratioDiff.GT(ratioThreshold) {
+		return sdk.ZeroUint(), sdk.ZeroUint(), types.ErrAsymmetricAdd
```

Comment:
@timlind what do you think of using a different error message for the ratio asymetric add error?

Reply:
Can do.